app/initializers/flash-messages.js

```diff
@@ -2,42 +2,16 @@ import Ember from 'ember';
 import config from '../config/environment';
 /* eslint-disable ember/new-module-imports */
 const { deprecate } = Ember;
-const merge = Ember.assign || Ember.merge;
 const INJECTION_FACTORIES_DEPRECATION_MESSAGE = '[ember-cli-flash] Future versions of ember-cli-flash will no longer inject the service automatically. Instead, you should explicitly inject it into your Route, Controller or Component with `Ember.inject.service`.';
-const addonDefaults = {
-  timeout: 3000,
-  extendedTimeout: 0,
-  priority: 100,
-  sticky: false,
-  showProgress: false,
-  type: 'info',
-  types: [
-    'success',
-    'info',
-    'warning',
-    'danger',
-    'alert',
-    'secondary'
-  ],
-  injectionFactories: [
-    'route',
-    'controller',
-    'view',
-    'component'
-  ],
-  preventDuplicates: false
-};
+import flashMessageOptions from 'ember-cli-flash/utils/flash-message-options';
 
 export function initialize() {
   const application = arguments[1] || arguments[0];
   const { flashMessageDefaults } = config || {};
   const { injectionFactories } = flashMessageDefaults || [];
-  const options = merge(addonDefaults, flashMessageDefaults);
+  const options = flashMessageOptions(flashMessageDefaults);
   const shouldShowDeprecation = !(injectionFactories && injectionFactories.length);
 
-  application.register('config:flash-messages', options, { instantiate: false });
-  application.inject('service:flash-messages', 'flashMessageDefaults', 'config:flash-messages');
```

Comment:
@fivetanley can you help me understand the change here? is this disabling the injection of the entire service, or just the config?

Reply:
@Dhaulagiri just the config. i grep'd the code base to look for usage, and this was the only one I could find. I don't think it's necessary any longer because it was used to inject onto the service.

Breaking down these two lines a bit:

```javascript
application.register('config:flash-messages', options, { instantiate: false });
application.inject('service:flash-messages', 'flashMessageDefaults', 'config:flash-messages');
```
`application.register` _defines_ a singleton thing.
`application.inject` _injects_ a singleton thing onto another singleton thing.

I tend to think about it like this:

```javascript
application.register(singletonThing, options, { instantiate: false });
application.inject(targetToInjectPropertyOnto, propertyName, propertyValueThatIsTheResultOfResolving);
```

so the tl;dr of that is that this initializer was injecting a `flashMessageDefaults` property onto each instance of the `flash-messsages` service. Because we're looking up the config directly in `addon/services/flash-messages.js`, and it already has a `flashMessageDefaults`, we don't need to inject it here.